### PR TITLE
clearer naming on bundle result events

### PR DESCRIPTION
**Pull request by segfaultdoc**

### CODE REVIEW

File: bundle.proto [hunk 1]
```diff
@@ -27,14 +27,33 @@ message Accepted {
...
+  }
+}
+
+// Indicates the bundle was dropped and therefore not forwarded to any validator.
+message RejectedV1 {
+  oneof reason {
+    StateAuctionBidRejected state_auction_bid_rejected = 1;
+    SimulationFailure simulation_failure = 2;
+    InternalError internal_error = 3;
+    string other = 4;
```
Comment (esemeniuc) [inline bundle.proto:56]: maybe set other to have higher id (eg 10) to allow for more reasons

File: bundle.proto [hunk 2]
```diff
@@ -81,31 +100,64 @@ message Processed {
...
 
 message BundleResult {
   // Bundle's Uuid.
   string bundle_id = 1;
 
   oneof result {
-    // Indicated accepted by the block-engine and forwarded to a jito-solana validator.
-    Accepted accepted = 2;
+    // Indicates accepted by the block-engine and forwarded to a jito-solana validator.
+    // Deprecated, start using forwarded_to_validator.
```
Comment (esemeniuc) [inline bundle.proto:125]:
```suggestion
    // Deprecated, replace with `forwarded_to_validator`.
```

File: bundle.proto [hunk 3]
```diff
@@ -81,31 +100,64 @@ message Processed {
...
+
+  oneof result {
+    // Rejected by the block-engine.
+    Rejected rejected = 2;
+
+    // Reached finalized commitment level.
+    Finalized finalized = 3;
+
+    // Reached a processed commitment level.
+    Processed processed = 4;
```
Comment (esemeniuc) [inline bundle.proto:156]: maybe reorder to be from lowest guarantee to highest guarantee?
eg.
rejected
dropped
processed
finalized

File: bundle.proto [hunk 4]
```diff
@@ -27,14 +27,35 @@ message Accepted {
...
+    // Some internal error occurred with the Block-Engine.
     InternalError internal_error = 4;
-    DroppedBundle dropped_bundle = 5;
+
+    // Indicates accepted by the block-engine and forwarded to a jito-solana validator.
+    // Deprecated, start using `other`.
+    DroppedBundle dropped_bundle = 5 [deprecated = true];
+
+    // Set to really high number so we can potentially add other fields before.
+    string other = 100;
```
Comment (buffalu) [inline bundle.proto:58]: can remove this. seems useless at first glance and will be another match statement in code for people reading this message

File: bundle.proto [hunk 5]
```diff
@@ -81,31 +107,44 @@ message Processed {
...
     Rejected rejected = 3;
+
     // Reached finalized commitment level.
     Finalized finalized = 4;
+
     // Reached a processed commitment level.
     Processed processed = 5;
-    // Was accepted and forwarded by the block-engine but never landed on-chain.
+
+    // Dropped from the mempool.
```
Comment (buffalu) [inline bundle.proto:144]: what are drop reasons?

Comment (buffalu) [reply bundle.proto:144]: nvmd the message explains that

Comment (buffalu) [inline bundle.proto:144]: whats the difference between rejected and dropped?

File: bundle.proto [hunk 6]
```diff
@@ -27,14 +27,35 @@ message Accepted {
   string validator_identity = 2;
 }
 
+message ForwardedToValidator {
+  // Slot at which bundle was forwarded.
+  uint64 slot = 1;
+
+  // Validator identity bundle was forwarded to.
```
Comment (esemeniuc) [inline bundle.proto:34]:
```suggestion
  // Validator identity for which the bundle was forwarded to.
```

File: bundle.proto [hunk 7]
```diff
@@ -52,9 +73,14 @@ message WinningBatchBidRejected {
 message StateAuctionBidRejected {
   // Auction's unique identifier.
   string auction_id = 1;
+
   // Bundle's simulated bid.
   uint64 simulated_bid_lamports = 2;
+
   optional string msg = 3;
+
+  // This auction's winning bundle bid.
```
Comment (esemeniuc) [inline bundle.proto:82]:
```suggestion
  // This auction's winning bid in lamports.
```

File: bundle.proto [hunk 8]
```diff
@@ -27,14 +27,35 @@ message Accepted {
...
 
+message ForwardedToValidator {
+  // Slot at which bundle was forwarded.
+  uint64 slot = 1;
+
+  // Validator identity bundle was forwarded to.
+  string validator_identity = 2;
+}
+
 // Indicates the bundle was dropped and therefore not forwarded to any validator.
```
Comment (esemeniuc) [inline bundle.proto:38]:
```suggestion
// Indicates the bundle was rejected by the Block-Engine and therefore not forwarded to any validator.
```

File: bundle.proto (hunk 5 above)
Comment (esemeniuc) [reply bundle.proto:144]: suggestion:
rejected means user did something wrong (which is user-correctable)
dropped means block engine could not process it due to some internal error

Comment (buffalu) [reply bundle.proto:144]: i like it. rejected = 4xx error code, dropped = 5xx error code

Comment (buffalu) [reply bundle.proto:144]: you have Status for streams tho, so not exactly equal. but maybe general guideline

File: bundle.proto [hunk 9]
```diff
@@ -27,14 +27,35 @@ message Accepted {
...
+
+    // Bundle failed simulation.
     SimulationFailure simulation_failure = 3;
+
+    // Some internal error occurred with the Block-Engine.
     InternalError internal_error = 4;
-    DroppedBundle dropped_bundle = 5;
+
+    // Indicates accepted by the block-engine and forwarded to a jito-solana validator.
+    // Deprecated, start using `other`.
```
Comment (buffalu) [inline bundle.proto:54]: use of other here seems weird. is other replacing dropped? why not keep Dropped around? it seems easier to conditionally switch on an enumeration vs a string

File: bundle.proto (hunk 7 above)
Comment (segfaultdoc) [reply bundle.proto:82]: could potentially accept bids in other tokens e.g. jitoSOL, usdc etc.

File: bundle.proto (hunk 9 above)
Comment (segfaultdoc) [reply bundle.proto:54]: `DroppedBundle` is useless, it has one string msg field. too many nested fields annoying so other is just flattening it

File: bundle.proto (hunk 5 above)
Comment (segfaultdoc) [reply bundle.proto:144]: maybe need to rename dropped, but dropped is representing bundles that were forwarded but did not land hence `DroppedReason` enum types. rejected bundles never got forwarded

**Pull request closed without merging**